### ISSUE TITLE
Step 10 fails!  Now what?

### ISSUE THREAD

**Issue by drasmu** [SharePoint/sp-dev-docs#1485]
Everything goes according to the directions above until I get to step 10.  My app builds, packages, and is successfully added to the "Apps for SharePoint" library in my App Catalog site.  But when I select the app to install the solution on a site, after a few minutes and a few page refreshes, the app remains "greyed out" and is not running.  Clicking the title displays text that says the solution "is initialized" but "isn't available".  If I switch back to "Classic View" it says " Sorry, something went wrong with adding the app. Click to retry."  But retrying has the same results.  Now what?  How do I troubleshoot and find out what went wrong?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: f46bf1b1-6910-2905-43ad-d24d88ef05e3
* Version Independent ID: f74fc1df-0ae6-6f5b-5008-9e95922faa9d
* Content: [Deploy your extension to SharePoint (Hello World part 3)](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/extensions/get-started/serving-your-extension-from-sharepoint)
* Content Source: [docs/spfx/extensions/get-started/serving-your-extension-from-sharepoint.md](https://github.com/SharePoint/sp-dev-docs/blob/master/docs/spfx/extensions/get-started/serving-your-extension-from-sharepoint.md)
* Service: **unspecified**
* GitHub Login: @spdevdocs
* Microsoft Alias: **spdevdocs**

**Comment by mpasarin** [comment]
Thanks for your feedback @drasmu 
From the Classic View, if you click on the "..." button you can access the app details.
In the app details page you'll be able to see the app install failure and get more logs from them.

Can you take a look at what went wrong there? We can work once we know what's the actual issue with the installation.
If that's inconclusive, could you then send us the site URL and the day you tried to install the app?

**Comment by drasmu** [comment]
I have tried many times, including again this morning. The details are as follows: 

Properties App Profile 
Version 	1.0.0.0 
Installed by 	Daryl Rasmussen 
Installed date 	Wednesday, March 21, 2018 
Errors 
Install Errors 	1 
Upgrade Errors 	0 
Runtime Errors 	0 (past 4 days) 

Clicking the hyperlink for the 1 "Install Errors" shows this: 

There were problems with the app web definition in the package. 
Details: Deployment failed in host web https://calgarymfrc.sharepoint.com for app modern-menu-styler-client-side-solution/431755f6-656d-4925-977a-d8f07860af61. System.ArgumentException: Required tag 'http://schemas.microsoft.com/sharepoint/:Elements' is missing from XML file '0564cffb-b096-4cd8-a368-10661da68950\elements.xml', found 'http://schemas.microsoft.com/sharepoint:Elements' instead. at Microsoft.SharePoint.Utilities.SPUtility.XsdValidateXml(XmlTextReader xmlStreamReader, String friendlyName, String pathXsdFile, String tagExpectedRootNode, Int32 desiredPathVersion, ValidationEventHandler xsdValEventHandler) at Microsoft.SharePoint.Utilities.SPUtility.XsdValidateXml(XmlTextReader xmlStreamReader, String friendlyName, String pathXsdFile, String tagExpectedRootNode, ValidationEventHandler xsdValEventHandler) at Microsoft.SharePoint.Administration.SPSolutionPackage.SolutionFile.ValidateFeatureXmlFile(String rootTagElement) at Microsoft.SharePoint.Administration.SPSolutionPackage.AddFeatureElementsCore(XmlNodeList nodeList, String strFeatureDirRelativeToCabFile, Guid featureId, Dictionary`2 filesAdded) at Microsoft.SharePoint.Administration.SPSolutionPackage.AddFeatureElements(XmlNode root, String strFeatureDirRelativeToCabFile, String strFeatureXmlFilename, Guid featureId) at Microsoft.SharePoint.Administration.SPSolutionPackage.WspSolutionFeature.ProcessFeatureXml() at Microsoft.SharePoint.Administration.SPSolutionPackage.InitSolutionFeatures(XmlNode root) at Microsoft.SharePoint.Administration.SPSolutionPackage.ProcessSolutionManifest() at Microsoft.SharePoint.Administration.SPSolutionPackage.Load() at Microsoft.SharePoint.Packaging.SPTargetWebDeploymentGroup.GenerateAndActivateUserSolutionFromPackage(SPWeb web, SPAppPackage appPackage, Guid productId, String solutionName, SPSolutionLevel solutionLevel) at Microsoft.SharePoint.Packaging.SPTargetWebDeploymentGroup.InstallOrUpgrade(SPSite site, Nullable`1& solutionId, Boolean& swapNeeded) 

Daryl

**Comment by drasmu** [comment]
Ok, so now that I know where to look for the error, it appears that the lack of a slash at the end of the url in the "Elements" tag in my elements.xml file is the problem.  I had <Elements xmlns="http://schemas.microsoft.com/sharepoint".  Changing this to <Elements xmlns="http://schemas.microsoft.com/sharepoint/"> and trying again...

**Comment by drasmu** [comment]
Yup, that seems to have fixed it.  Too bad the error isn't trapped much earlier.  The solution works fine without the final slash in the Elements tag when in the debugger, but at least I know now where to find error(s) if they happen later in the development cycle.  Good for others to know the troubleshooting steps:  1.  Switch to Classic view.  2.  Click the ellipses.  3.  Choose Details.  4.  Click the number beside "Install Errors" to get more details about what is going on.
